NPE loading notebook written in older format

**Describe the bug**
Loading an old notebook fails with a NullPointerException.

```
jakarta.json.bind.JsonbException: Internal error: Cannot invoke "jakarta.json.JsonString.getString()" because the return value of "org.eclipse.parsson.JsonObjectBuilderImpl$JsonObjectImpl.getJsonString(String)" is null
        at org.eclipse.yasson.internal.DeserializationContextImpl.deserializeItem(DeserializationContextImpl.java:142)
        at org.eclipse.yasson.internal.DeserializationContextImpl.deserialize(DeserializationContextImpl.java:127)
        at org.eclipse.yasson.internal.JsonBinding.deserialize(JsonBinding.java:55)
        at org.eclipse.yasson.internal.JsonBinding.fromJson(JsonBinding.java:78)
        at org.jtaccuino.app.common.NotebookPersistence.of(NotebookPersistence.java:54)
        at org.jtaccuino.app.studio.actions.OpenAction.handle(OpenAction.java:45)
        at org.jtaccuino.app.studio.actions.OpenAction.handle(OpenAction.java:26)
        at javafx.base@24/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:86)
        at javafx.base@24/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:232)
        at javafx.base@24/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:189)
        at javafx.base@24/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at javafx.base@24/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at javafx.base@24/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base@24/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base@24/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base@24/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base@24/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base@24/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at javafx.base@24/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:49)
        at javafx.base@24/javafx.event.Event.fireEvent(Event.java:199)
        at javafx.graphics@24/javafx.scene.Node.fireEvent(Node.java:9005)
        at javafx.controls@24/javafx.scene.control.Button.fire(Button.java:203)
        at javafx.controls@24/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(ButtonBehavior.java:207)
        at javafx.controls@24/com.sun.javafx.scene.control.inputmap.InputMap.handle(InputMap.java:274)
        at javafx.base@24/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(CompositeEventHandler.java:247)
        at javafx.base@24/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(CompositeEventHandler.java:80)
        at javafx.base@24/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:232)
        at javafx.base@24/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(EventHandlerManager.java:189)
        at javafx.base@24/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(CompositeEventDispatcher.java:59)
        at javafx.base@24/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:58)
        at javafx.base@24/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base@24/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base@24/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base@24/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(BasicEventDispatcher.java:56)
        at javafx.base@24/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(EventDispatchChainImpl.java:114)
        at javafx.base@24/com.sun.javafx.event.EventUtil.fireEventImpl(EventUtil.java:74)
        at javafx.base@24/com.sun.javafx.event.EventUtil.fireEvent(EventUtil.java:54)
        at javafx.base@24/javafx.event.Event.fireEvent(Event.java:199)
        at javafx.graphics@24/javafx.scene.Scene$MouseHandler.process(Scene.java:4005)
        at javafx.graphics@24/javafx.scene.Scene.processMouseEvent(Scene.java:1934)
        at javafx.graphics@24/javafx.scene.Scene$ScenePeerListener.mouseEvent(Scene.java:2763)
        at javafx.graphics@24/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.get(GlassViewEventHandler.java:352)
        at javafx.graphics@24/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.get(GlassViewEventHandler.java:254)
        at javafx.graphics@24/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(QuantumToolkit.java:424)
        at javafx.graphics@24/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(GlassViewEventHandler.java:386)
        at javafx.graphics@24/com.sun.glass.ui.View.handleMouseEvent(View.java:560)
        at javafx.graphics@24/com.sun.glass.ui.View.notifyMouse(View.java:946)
        at javafx.graphics@24/com.sun.glass.ui.mac.MacView.notifyMouse(MacView.java:128)
Caused by: java.lang.NullPointerException: Cannot invoke "jakarta.json.JsonString.getString()" because the return value of "org.eclipse.parsson.JsonObjectBuilderImpl$JsonObjectImpl.getJsonString(String)" is null
        at org.eclipse.parsson.JsonObjectBuilderImpl$JsonObjectImpl.getString(JsonObjectBuilderImpl.java:264)
        at org.jtaccuino.app.common.internal.IpynbFormat$CellDeserializer.deserialize(IpynbFormat.java:180)
        at org.jtaccuino.app.common.internal.IpynbFormat$CellDeserializer.deserialize(IpynbFormat.java:173)
        at org.eclipse.yasson.internal.deserializer.UserDefinedDeserializer.deserialize(UserDefinedDeserializer.java:69)
        at org.eclipse.yasson.internal.deserializer.UserDefinedDeserializer.deserialize(UserDefinedDeserializer.java:26)
        at org.eclipse.yasson.internal.deserializer.CollectionDeserializer.deserialize(CollectionDeserializer.java:49)
        at org.eclipse.yasson.internal.deserializer.CollectionInstanceCreator.deserialize(CollectionInstanceCreator.java:61)
        at org.eclipse.yasson.internal.deserializer.CollectionInstanceCreator.deserialize(CollectionInstanceCreator.java:37)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:85)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:34)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:46)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:26)
        at org.eclipse.yasson.internal.deserializer.ContextSwitcher.deserialize(ContextSwitcher.java:36)
        at org.eclipse.yasson.internal.deserializer.ContextSwitcher.deserialize(ContextSwitcher.java:22)
        at org.eclipse.yasson.internal.deserializer.JsonbCreatorDeserializer.deserialize(JsonbCreatorDeserializer.java:86)
        at org.eclipse.yasson.internal.deserializer.JsonbCreatorDeserializer.deserialize(JsonbCreatorDeserializer.java:38)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:85)
        at org.eclipse.yasson.internal.deserializer.PositionChecker.deserialize(PositionChecker.java:34)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:46)
        at org.eclipse.yasson.internal.deserializer.NullCheckDeserializer.deserialize(NullCheckDeserializer.java:26)
        at org.eclipse.yasson.internal.DeserializationContextImpl.deserializeItem(DeserializationContextImpl.java:138)
        ... 47 more
```

**To Reproduce**
Steps to reproduce the behavior:
1. Open a Notebook written with Notebook file format prior to v4.5
2. Poof

**Expected behavior**
Notebook is loaded successfully.

**OS (please complete the following information):**
 - MacOS, maybe others

--- a/app/src/main/java/org/jtaccuino/app/common/internal/IpynbFormat.java
+++ b/app/src/main/java/org/jtaccuino/app/common/internal/IpynbFormat.java
@@ -177,7 +177,10 @@ public record IpynbFormat(Map<String, Object> metadata, int nbformat, int nbform
         public Cell deserialize(JsonParser parser, DeserializationContext ctx, Type rtType) {
             var o = parser.getObject();
             var type = o.getString("cell_type");
-            var id = o.getString("id");
+            var id = o.containsKey("id")
+                    // Mandatory since nbformat v4.5. Did not exist prior to that
+                    ? o.getString("id")
+                    : UUID.randomUUID().toString();
             var source = multilineOrArrayToString(o.get("source"));
             var metadata = (Map<String, Object>) extractJsonData(o.getJsonObject("metadata"));
             return switch (type) {

